### PR TITLE
framework/task_manager: Remove unnecessary TM_FREE and Modify its TC

### PR DESCRIPTION
**Pull request by jeongchanKim**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f06ad90fd77085c5ac1e06bd3542945778daf242] - Code Rule Check OK.

File: framework/src/task_manager/task_manager_unicast.c [hunk 1]
```diff
@@ -108,7 +108,6 @@ int task_manager_unicast(int handle, tm_unicast_msg_t *send_msg, tm_unicast_msg_
 		}
 	}
 
-	TM_FREE(((tm_unicast_internal_msg_t *)request_msg.data)->msg);
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager_unicast.c]: Then, who frees this? Please confirm it.

File: framework/src/task_manager/task_manager_unicast.c [hunk 2]
```diff
@@ -136,9 +134,7 @@ int task_manager_reply_unicast(tm_unicast_msg_t *reply_msg)
 		return TM_OUT_OF_MEMORY;
 	}
 	data->msg_size = reply_msg->msg_size;
-	memcpy(data->msg, reply_msg->msg, reply_msg->msg_size);
-
-	handle = taskmgr_get_handle_by_pid(getpid());
+	memcpy(data->msg, reply_msg->msg, reply_msg->msg_size + 1);
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager_unicast.c]: Please don't make a decision by task manager itself. User gives msg_size. Don't do "+ 1" even if it is wrong.

Comment (sunghan-chang) [reply framework/src/task_manager/task_manager_unicast.c]: If someone gives rigth value, buffer size 10 including null and user gives 10 as size, at that case this causes invalid memory access.

File: framework/src/task_manager/task_manager_unicast.c (hunk 1 above)
Comment (jeongchanKim) [reply framework/src/task_manager/task_manager_unicast.c]: User should free this. I will update the explanation for this API.

**Comment by seinfra** [comment]
Target : [dd65ea9b47af74d376758a443c7a5c1cabc3d00f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [dd65ea9b47af74d376758a443c7a5c1cabc3d00f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0f3cfe5f429408d32c733c33a45710718088ae39] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0f3cfe5f429408d32c733c33a45710718088ae39] - Code Rule Check (C++) OK.

**Pull request merged**